### PR TITLE
[RA] Complete balance patch (1.6) for the next playtest/release.

### PR DESCRIPTION
**Pull request by SoScared**
**To skip the abstract you can safely jump down straight to the changes themselves (see bold text below).**

I initially held off these changes from #13640 in anticipation of the HitShape/UnitStance features in order to better evaluate its balance implications. A few changes listed below are adjustments from the current playtest but the majority represents the full experience coming out of the playtest maps concluded with the so-called v1.6 map playtest series: http://www.sleipnirstuff.com/forum/viewtopic.php?f=83&t=20017

The map playtest v1.6 was distributed with ca 17.000 (!) map downloads with [1.6a](https://pastebin.com/7HPdCryT) and [1.6b](https://pastebin.com/SvX4rBAm) (latter version [still up on the resource site](http://resource.openra.net/maps/uploader/78/)) after an aggressive distribution campaign, resulting in maps being redistributed within the community itself mostly within a period of ca 2-3 months. The download count doesn't tell us really the quality of the changes itself but shows the balance changes has been accessible, redistributed and familiarized with the larger community for a long period of time. This made it possible to observe hundreds upon hundreds of games periodically which made it possible to track the progression of both 1v1s and team games and on how players dealt with the changes over time.

As with my previous balance PR's I will refer to the experience drawn from the playtesting itself, not delving too much into discussion about other changes or experiences drawn from sources outside this particular playtest series. This outlines my attitude towards playtesting in the past where I would often dismiss concerns on past balance PRs from other community members that contrasted the experiences drawn from the playtesting itself. These often originated from other sources or just concerns in general drawn from general insight of the game. This might have come across as unfair to some, especially now that the community has recently split up into multiple playtest series, often overlapping certain tests of changes but at the end of the day that's the only way I can personally guarantee the experience gathered from the map playtests will accurately reflect that of the upcoming playtest/release.

- - - - 

This balance patch is a culmination of pretty much everything I've worked for up to this point and was part of a long-term 2-punch-combo plan starting with #12739 for the current release. The aim was to make way for a broad yet safe path for the RA mod to provide more fun, colorful and exciting games for everybody while at the same time slowly expanding some options for our most dedicated of players in the community. No doubt this playtest series managed to achieved that and is my proudest work in my time with OpenRA, especially in a time with growth in both the general player base and the competitive arena, challenged with more viewpoints and opinions than ever putting on an ever growing pressure to offer more of what the RA mod already provides.

- - - - - - - - - - - -

**For anyone not having followed up on the playtest series previously, don't let the long list of changes intimidate. In a nutshell the changes are more akin to tuning an equalizer, and with few exceptions has been quality checked over a very long period of time.**

**The proposal includes the following changes**:

- - - - - - - - - - - -

**VEHICLES**

**Medium/Heavy/Mammoth Tank**
Damage vs None: 30%, up from 20%
Damage vs Heavy: 115%, up from 100%

**Heavy Tank**
HP: 600, up from 550

**Mammoth Tank***
Added Cloak Detection range: 6

**Artillery***
Innaccuracy: 1c938, down fromk 2c256 (playtest)
Damage: 230, down from 240 (playtest)
Damage vs None: 80, down from 90 (playtest)
Damage vs Wood: 35, down from 40 (playtest)

**Phase/Chrono/Tesla Tank**
Production time: 28 sec, down from 33 sec

**Chrono/Tesla Tank**
HP: 450, up from 400

**MCV**
Speed: 71, down from 85

**Mobile Radar Jammer, Mobile Gap Generator***
Speed: 99, up from 85

**Comments**:
* **The cloak detection for the Mammoth Tank** is part of a group of changes aimed to make the use of the Phase Transport more deliberate. All changes involving cloak detection in this list is part of that package and the standardization of cloak detection range for all defensive structures in the playtest preceded this move. 
* **The Artillery weapons change** is a further tweak from that of the playtest. The increase of innaccuracy felt slightly overboard and so I pulled it back a bit and compensated by pulling back some of its damage output. 
* **The speed upgrade for the MRJ and MGG** is the only feature from now a previously planned round of playtest maps that despite never having seen the light of day I feel is worth bringing along, at the very least towards the official playtest. As with the obscure Mammoth Tank pre-release-20161019 a speed upgrade will likely make the units more applicable. It's not a change I'm committed to in this PR but I highly recommend giving it a go.

- - - - - - - - - - - -

**INFANTRY**

**Medic**
HP: 60, down from 80

**Shock Trooper**
Cost: $300, down from $400
HP: 50, down from 60
Damage vs Wood 75%, vs Heavy 60%

**Tanya**
Added Cloak Detection range: 5

**Grenadier**
Damage vs None: 60% (up from 50%)

- - - - - - - - - - - -

**AIRCRAFT**

**MiG/Longbow**
Production time: 42 sec, down from 48 sec

**Chinook**
Speed: 128, up from 112
Vision: 8, down from 10
HP: 140, up from 120
Land/Liftoff speed: 0c58, down from 0c100

- - - - - - - - - - - -

**NAVAL**

**Cruiser**
Production time: 48 sec, down from 58 sec

**Missile Sub**
Production time: 42 sec, down from 48 sec

- - - - - - - - - - - -

**STRUCTURES**

**Barracks**
Cost: $500, up from $400

**War Factory***
Build-time reduction caps at 50% with 4 structures (100, 75, 60, 50), down from of 7 (100, 85, 75, 65, 60, 55, 50)

**Radar Dome**
Power-down disables cloak detection

**Soviet Tech Center***
HP: 800, up from 600

**Refinery**
Vision range: 5, down from 6

**Comments**: 
* **The War Factory production cap** is dealt with at _the bottom of this post_ as it deals with overall game design issues and requires a broader comment. 
* **The Soviet Tech Center HP increase** is in response to the new hitshape changes as the structure is one of the big losers in the AoE department. The parabombs flattens the structure. The HP increase will be quite noticeable in-game balance wise but as-is I don't see any other way to approach the issue practically.

- - - - - - - - - - - -

**DEFENSIVE STRUCTURES**

**AAGun/SAM**
Removed cloak detection

**SAM**
Cost: $700, down from $750

**Pillbox/C.Pillbox/Flame Tower**
Damage vs Light: 50%, down from 60%

**Walls**
HP: 400, down from 500

**Sandbag/Fence**
HP: 150, down from 300

- - - - - - - - - - - -

**SUPPORT POWERS**

**Paradrops**
Chargetime: 5 min, down from 6 min. 

**Parabombs**
Chargetime: 5 min, down from 6 min. 

**Spy Plane**
Chargetime: 2 min 30sec, down from 3 min.

- - - - - - - - - - - -

**CIVILIAN STRUCTURES**

**Forward Command**
Capturable, grants Base Provider Trait (8c0)
HP: 800
Powercost: 0
Vision Range: 5
Takes Engineer 2x normal cap time

**Technology Center***
Renamed to Communications Center
Capturable, grants 10c0 vision range
HP: 600

**Comments**:
The remaining civilian structures coming alive is one of my personal favorite changes, I had a lot of fun experimenting with these on few maps. The Technology Center is renamed to Communications Center in order to distinguish it from the buildable Tech Center. Originally the Technology Center building went by so many different names and uses in CnC and RA, almost as a running joke, so renaming it shouldn't be an issue.

- - - - - - - - - - - -

**War Factory, new production cap**

One of the above changes needs to have the developer team on-board as it does essentially deal with questions regarding game design. The War Factory production capped at 4 instead of 7 breaks with all other build queues and essentially creates a two-tiered queue system. Practically the WF queue in relation to the others looks like this:
red alert blackground 1920x1200-2
With the 3 middle values scooped out the War Factory is allowed to reduce its production time with 25% on the second War Factory.

I don't remember who came up with the idea but I decided to bring it aboard early on and throw it out at the first sign of trouble. The War Factory being a $2000 structure I figured there was a chance it would add to the game without disrupting too much. To my surprise it basically under-performed, being phased out quickly as an early-to-mid-game option after just a few days of players attempting to abuse the vehicle production gain as a build-order opener. The geniality of the feature became apparent at the later stages in the games, ca. 15-20 minutes into the game after the tech-tree had fully developed both in 1v1s and team games. Practically there were more options presented on the table late-game with quicker transitions between types of vehicles and in combination with the new T3 custom build times and HP upgrades (phase/chrono/tesla tank) represented in this PR we saw players playing around with more of these units and often helping players close out games in a more colorful fashion instead of grinding the beaten player down in long drawn-out stalemate situations. It never felt over-board except perhaps in games with people playing around having their fun but at the expense of winning the game against an equal opponent.

The standard 7-point production queues themselves being a bit obscure to most players makes the deviation of a 4-point War Factory seemingly even more so but I would ask you to consider this - it never came up as an issue in any games I observed in the entire time. In fact players were discussing among themselves about the values given that the WF now had a distinct different production cap value. Those values themselves being simply a derivative of the original 7-point value (as visualized above) made the WF 4-point value easy and quick to learn and in return ironically fostered awareness on what the standard queue values really are.

It's my intention to sell this point across to the developers to add this deviation to the game for the great fun, the rewards and vast potential this feature adds to the experience. This playtest/release has with it a great many changes to standardize certain elements for sake of ease of understanding including building vision range and and cloak detection for the defensive structures. Those deviations served no purpose. This one does and I'd anticipate it to be one of the most well received balance features with the next release.

- - - - - - - - - - - -

As mentioned initially all of these changes has been playtested at an unprecedented rate and, as with the previous balance PRs, can be expected to take aboard changes that are predictable and expected to function in the release as promised.

### CODE REVIEW

File: mods/ra/rules/civilian.yaml [hunk 1]
```diff
@@ -70,18 +70,31 @@ FCOM:
 		Dimensions: 2,3
 		LocalCenterOffset: 0,-512,0
 	Valued:
-		Cost: 2000
+		Cost: 1
```
Comment (abcdefg30) [inline mods/ra/rules/civilian.yaml]: Is this change still needed as you already disable bounties on the structure?

File: mods/ra/rules/civilian.yaml [hunk 2]
```diff
@@ -70,18 +70,31 @@ FCOM:
...
 	Health:
-		HP: 400
+		HP: 800
 	Armor:
 		Type: Wood
 	Tooltip:
 		Name: Forward Command
+	GivesBounty:
```
Comment (abcdefg30) [inline mods/ra/rules/civilian.yaml]: I think it'd be better to remove this entirely using `-GivesBounty:` instead of setting the `Percentage` to 0.

File: mods/ra/rules/civilian.yaml [hunk 3]
```diff
@@ -353,16 +366,25 @@ MISS:
 		Dimensions: 3,3
 		LocalCenterOffset: 0,-512,0
 	Valued:
-		Cost: 2000
+		Cost: 1
```
Comment (abcdefg30) [inline mods/ra/rules/civilian.yaml]: Same question here.

File: mods/ra/rules/civilian.yaml [hunk 4]
```diff
@@ -353,16 +366,25 @@ MISS:
...
 	Health:
-		HP: 400
+		HP: 600
+	GivesBounty:
```
Comment (abcdefg30) [inline mods/ra/rules/civilian.yaml]: And same suggestion here.

File: mods/ra/rules/civilian.yaml [hunk 5]
```diff
@@ -353,16 +366,25 @@ MISS:
...
+	RevealsShroud@GAPGEN:
+		Range: 6c0
 	Armor:
 		Type: Wood
 	Tooltip:
 		Name: Technology Center
 	WithBuildingBib:
+	ExternalCapturable: 
+		ExternalCapturableBar:
+		EngineerRepairable:
```
Comment (abcdefg30) [inline mods/ra/rules/civilian.yaml]: The indentation on those two here looks wrong.

File: mods/ra/rules/vehicles.yaml [hunk 6]
```diff
@@ -213,6 +213,9 @@ V2RL:
 		VisualBounds: 44,38,0,-4
 	ProducibleWithLevel:
 		Prerequisites: vehicles.upgraded
+	DetectCloaked:
+		Range: 6c0
+
```
Comment (abcdefg30) [inline mods/ra/rules/vehicles.yaml]: Superfluous newline.

File: mods/ra/weapons/other.yaml [hunk 7]
```diff
@@ -29,7 +29,9 @@ FireballLauncher:
 		Speed: 250
 		TrailImage: fb2
 		Image: FB1
-
+	Warhead@1Dam: SpreadDamage
+		Versus:
+			Light: 50
 Flamer:
```
Comment (abcdefg30) [inline mods/ra/weapons/other.yaml]: I think you're missing a newline above this line.

File: mods/ra/rules/civilian.yaml (hunk 1 above)
Comment (SoScared) [inline mods/ra/rules/civilian.yaml]: Neither 'givebounty', 'cost' or 'valued' was neccessary and depended on each other in a closed loop. Removed them all (as done with Oil Derric/Hospital).

File: mods/ra/rules/civilian.yaml (hunk 2 above)
Comment (SoScared) [inline mods/ra/rules/civilian.yaml]: ^fixed

File: mods/ra/rules/civilian.yaml (hunk 3 above)
Comment (SoScared) [inline mods/ra/rules/civilian.yaml]: ^fixed

File: mods/ra/rules/civilian.yaml (hunk 4 above)
Comment (SoScared) [inline mods/ra/rules/civilian.yaml]: ^fixed

File: mods/ra/rules/civilian.yaml (hunk 5 above)
Comment (SoScared) [inline mods/ra/rules/civilian.yaml]: Too fond of copy/paste. Fixed.

File: mods/ra/rules/vehicles.yaml (hunk 6 above)
Comment (SoScared) [inline mods/ra/rules/vehicles.yaml]: Fixed.

File: mods/ra/weapons/other.yaml (hunk 7 above)
Comment (SoScared) [inline mods/ra/weapons/other.yaml]: Fixed.

**Review by abcdefg30 — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/65d025bd3f117cecc557cde67a3de08b01378b75...fb63007c02aaaa24794bdec0bd4342cdfcacee5d
